pkg/fsutil/dir.go

```diff
@@ -99,3 +100,27 @@ func GetIntraDir(pattern string, depth, length int) string {
 	}
 	return intraDir
 }
+
+// GetIntraDirID returns a string that can be added to filepath.Join to implement directory depth with
+// sequential numbers.
+// Where the pattern length is more than provided length, the maxID for length is prepended to the result, then
+// GetIntraDir is called to break the pattern into smaller chunks of the provided length.
+// eg: for an id of 1234 and a length of 3, the result is 1000/001
+// for an id of 12345678 and length 3, the result is 1000000/012/345
+func GetIntraDirID(id int, length int) string {
+	ret := ""
+	thisLen := length
+	max := int(math.Pow10(thisLen))
+	for id >= max {
+		ret = filepath.Join(fmt.Sprintf("%d", max))
```

Comment:
Probably `ret = fmt.Sprintf("%d", max) ` ?
As pointed out by go critic the join doesnt do anything with a single element